src/unix/sock_ctrl_msg.rs

```diff
@@ -253,8 +218,8 @@ unsafe fn raw_recvmsg(
                 // alignment. If these fds can not be stored in `in_fds` buffer, then all the control
                 // data must be dropped to insufficient buffer space for returning them to outer
                 // scope. This might be a sign of incorrect protocol communication.
-                for fd_offset in 0..fds_count {
-                    let raw_fds_ptr = CMSG_DATA(cmsg_ptr);
+                for fd_offset in 0..fds_to_be_copied_count {
+                    let raw_fds_ptr: *mut RawFd = CMSG_DATA(cmsg_ptr).cast();
```

Comment:
As discussed, this change should be in a single commit, explaining what is happening, so it will be more clear that we need help with this

Reply:
@uran0sH okay, but the point here is why:

```
When running send_more_recv_less2 or send_more_recv_less1 on macos,
cmsg.cmsg_len is 28 and CMSG_LEN(0) is 12, which cause fds_count is 4.
```

so why this is happening?

Reply:
> @uran0sH okay, but the point here is why:
> 
> ```
> When running send_more_recv_less2 or send_more_recv_less1 on macos,
> cmsg.cmsg_len is 28 and CMSG_LEN(0) is 12, which cause fds_count is 4.
> ```
> 
> so why this is happening?

I think it's the same as https://github.com/rust-vmm/vmm-sys-util/pull/245#discussion_r2209055308. The value of align causes it

Reply:
@uran0sH okay, but for the future, the "why" is the important stuff to mention in commit description, PR, etc.
We don't want each people reviewing or looking later at the commit/code, going into the details to understand the reason.